common/forks/forks.go

```diff
@@ -0,0 +1,45 @@
+package forks
+
+import (
+	"math/big"
+	"sort"
+
+	"github.com/scroll-tech/go-ethereum/params"
+)
+
+// CollectSortedForkHeights returns a sorted set of block numbers that one or more forks are activated on
+func CollectSortedForkHeights(config *params.ChainConfig) ([]uint64, map[uint64]bool) {
+	forkHeightsMap := make(map[uint64]bool)
+	for _, fork := range []*big.Int{
+		config.HomesteadBlock,
+		config.DAOForkBlock,
+		config.EIP150Block,
+		config.EIP155Block,
+		config.EIP158Block,
+		config.ByzantiumBlock,
+		config.ConstantinopleBlock,
+		config.PetersburgBlock,
+		config.IstanbulBlock,
+		config.MuirGlacierBlock,
+		config.BerlinBlock,
+		config.LondonBlock,
+		config.ArrowGlacierBlock,
+		config.ArchimedesBlock,
+		config.ShanghaiBlock,
+	} {
+		if fork == nil {
+			continue
+		} else if height := fork.Uint64(); height != 0 {
+			forkHeightsMap[height] = true
+		}
+	}
+
+	var forkHeights []uint64
+	for height := range forkHeightsMap {
+		forkHeights = append(forkHeights, height)
+	}
+	sort.Slice(forkHeights, func(i, j int) bool {
+		return forkHeights[i] < forkHeights[j]
+	})
+	return forkHeights, forkHeightsMap
```

Comment:
`forkHeightsMap` is an unused return value.

Reply:
https://github.com/scroll-tech/scroll/pull/1174/files/e608c6bb3edea37719f3a00af909c45ee0de2664#diff-988037fb2a3b2ec2f89c5a3e6be65cba29cf2f45410c8a5214284982fa1d73d9R70 batch proposer does

Reply:
Oh, I see. Thanks.